tests/main/interfaces-snap-interfaces-requests-control/task.yaml

```diff
@@ -40,8 +40,8 @@ execute: |
 
     echo "Ensure AppArmor Prompting experimental feature can be enabled where possible"
     # prompting is disabled everywhere but the Ubuntu systems
-    # TODO on Ubuntu releases < 24.04 we need the snapd snap for testing
-    if ! os.query is-ubuntu || os.query is-ubuntu-lt 24.04 || os.query is-core ; then
+    # on Ubuntu releases < 22.04 the kernel does not support prompting
+    if ! os.query is-ubuntu || os.query is-ubuntu-lt 22.04 || os.query is-core ; then
         not snap set system experimental.apparmor-prompting=true >& err.out
```

Comment:
This is now failing on [debian-not-req](https://github.com/canonical/snapd/actions/runs/10623774678/job/29462432549?pr=14294#logs). It could because it is now using the vendored apparmor from the snapd snap

Reply:
If I understand the apparmor prompting checks correctly, it looks like debian is passing both the kernel version and apparmor version checks.

Reply:
hmm I wonder what to do about it, @olivercalder @pedronis what's the take on prompting support on other systems?

Reply:
I think we have a problem with prompting on Sid, clearly the kernel supposedly supports that, but it is non functional:
```
Aug 30 12:17:46 aug301205-227006 snapd[27725]: daemon.go:353: adjusting startup timeout by 55s (pessimistic estimate of 30s plus 5s per snap)
Aug 30 12:17:46 aug301205-227006 snapd[27725]: backends.go:58: AppArmor status: apparmor is enabled but some kernel features are missing: dbus, network
Aug 30 12:17:46 aug301205-227006 snapd[27725]: ifacemgr.go:214: failed to start interfaces requests manager: cannot register prompting listener: kernel does not support apparmor notifications
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: checking internal apparmor_parser candidate at /snap/snapd/x1/usr/lib/snapd/apparmor_parser
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: will consider standby after: 5s
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: Setting up sd_notify() watchdog timer every 2m30s
Aug 30 12:17:46 aug301205-227006 systemd[1]: Started snapd.service - Snap Daemon.
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: activation done in 185ms
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: Next refresh scheduled for 2024-08-31T11:31:35Z.
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: Catalog refresh starting now; next scheduled for 2024-08-31 16:59:04.186125898 +0000 UTC m=+103277.494156715.
Aug 30 12:17:46 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: Catalog refresh skipped when testing is enabled
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27777;uid=0;socket=/run/snapd.socket; GET /v2/system-info 14.501957ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27778;uid=0;socket=/run/snapd.socket; GET /v2/system-info 30.531352ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27777;uid=0;socket=/run/snapd-snap.socket; GET /v2/system-info 5.681345ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27823;uid=0;socket=/run/snapd.socket; GET /v2/system-info 9.408154ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27822;uid=0;socket=/run/snapd.socket; GET /v2/system-info 16.486108ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27822;uid=0;socket=/run/snapd-snap.socket; GET /v2/system-info 6.052376ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27865;uid=0;socket=/run/snapd.socket; GET /v2/system-info 6.159435ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27866;uid=0;socket=/run/snapd.socket; GET /v2/system-info 12.72486ms 200
Aug 30 12:17:47 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27865;uid=0;socket=/run/snapd-snap.socket; GET /v2/interfaces/requests/prompts 239.826µs 500
Aug 30 12:17:48 aug301205-227006 snapd[27725]: logger.go:99: DEBUG: pid=27945;uid=0;socket=/run/snapd.socket; GET /v2/connections?select=all 554.144µs 200
google-distro-1:debian-sid-64 .../tests/main/interfaces-snap-interfaces-requests-control# snap connections
Interface                         Plug                                         Slot                               Notes
home                              jq:home                                      :home                              -
snap-interfaces-requests-control  api-client:snap-interfaces-requests-control  :snap-interfaces-requests-control  manual
google-distro-1:debian-sid-64 .../tests/main/interfaces-snap-interfaces-requests-control#  api-client --socket /run/snapd-snap.socket /v2/interfaces/requests/prompts                                                                                                                                                                                                                                                                                                                     
{"type":"error","status-code":500,"status":"Internal Server Error","result":{"message":"Apparmor Prompting is not running"}}
```

I'll disable the test on Sid